pkg/build/builder/daemonless.go

```diff
@@ -309,10 +317,84 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 
 func generateTransientMounts() []string {
 	mounts := []string{}
+	mounts = appendRHSMMount(defaultMountStart, mounts)
+	mounts = appendETCPKIMount(defaultMountStart, mounts)
+	mounts = appendRHRepoMount(defaultMountStart, mounts)
 	mounts = appendCATrustMount(mounts)
 	return mounts
 }
 
+func appendRHRepoMount(pathStart string, mounts []string) []string {
+	path := filepath.Join(pathStart, repoFile)
+	st, err := os.Stat(path)
+	if err != nil {
+		// since the presence of the repo file is not a given, we won't log this a V(0)
+		log.V(4).Infof("Failed to stat %s, falling back to the Red Hat yum repository configuration in the build's base image. Error: %v", path, err)
+		return mounts
+	}
+	if !st.Mode().IsRegular() {
+		// if the file is there, but an unexpected type, then always have log show up via V(0)
+		log.V(0).Infof("Falling back to the Red Hat yum repository configuration in the build's base image: %s secrets location %s is a directory.", repoFile, path)
+		return mounts
+	}
+
+	// Add a bind of repo file, to pass along anything that the runtime mounted from the node
+	log.V(0).Infof("Adding transient rw bind mount for %s", path)
+	tmpDir, err := ioutil.TempDir("/tmp", repoFile+"-copy")
+	if err != nil {
+		log.V(0).Infof("Falling back to the Red Hat yum repository configuration in the base image: failed to create tmpdir for %s secret: %v", repoFile, err)
+		return mounts
+	}
+	fs := s2ifs.NewFileSystem()
+	err = fs.Copy(path, filepath.Join(tmpDir, repoFile), map[string]string{})
+	if err != nil {
+		log.V(0).Infof("Falling back to the Red Hat yum repository configuration in the base image: failed to copy %s secret: %v", repoFile, err)
+		return mounts
+	}
+	mounts = append(mounts, fmt.Sprintf("%s:/run/secrets/%s:rw,nodev,noexec,nosuid", filepath.Join(tmpDir, repoFile), repoFile))
+	return mounts
+}
+
+func coreAppendSecretLinksToDirs(pathStart, pathEnd string, mounts []string) []string {
+	path := filepath.Join(pathStart, pathEnd)
+	st, err := os.Stat(path)
+	if err != nil {
+		// since the presence of dir secret is not a given, we won't log this a V(0)
+		log.V(0).Infof("Red Hat subscription content will not be available in this build: failed to stat directory %s: %v", path, err)
+		return mounts
+	}
+	if !st.IsDir() && st.Mode()&os.ModeSymlink == 0 {
```

Comment:
space for readability? (looks like we are doing a bitwise AND here)

```suggestion
	if !st.IsDir() && (st.Mode() & os.ModeSymlink == 0) {
```

go fmt may remove my parens in the suggestion

Reply:
go fmt leaves the parens but removes the spaces you added @adambkaplan 

`if !st.IsDir() && (st.Mode()&os.ModeSymlink == 0) {`

that ^^ will be part of next push